resources/files/windows/PuppetBolt/PuppetBoltFunctions.ps1

```diff
@@ -0,0 +1,1639 @@
+        [int]     $ConnectTimeout,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [bool]    $TTY=$false,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $Tmpdir,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [bool]    $Color=$true,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [BoltOutputLevel]  $OutputLevel
+    )
+    begin {
+        $_flags = Convert-BoltParams @PSBoundParameters
+    }
+    process {
+        return Invoke-BoltInternal -CommandString "script run $ScriptLocation" @PSBoundParameters
+    }
+}
+
+function Read-BoltPlans {
+    <#
+    .SYNOPSIS
+    Read all bolt plans available to Puppet Bolt
+    .DESCRIPTION
+    This function will execute 'bolt plan show <plan>' to list available plans or details
+    on a specific plan. Puppet Bolt (https://puppet.com/products/bolt) is an agentless automation
+    solution for running ad-hoc tasks and operations on remote targets
+    .PARAMETER PlanName
+    Name of the bolt plan to show details on
+    .PARAMETER ModulePath
+    List of directories containing modules, separated by ';'
+    .PARAMETER BoltDir
+    Specify what Boltdir to load config from (default: autodiscovered from current working dir)
+    .PARAMETER ConfigFile
+    Specify where to load config from (default: ~/.puppetlabs/bolt/bolt.yaml)
+    .PARAMETER Color
+    Whether to show output in color
+    .PARAMETER OutputLevel
+    Level of output logging to display (defaults to no output). Must be one of:
+    * 'Verbose' only verbose logging
+    * 'Debug' both verbose and debug logging
+    * 'Trace' all of verbose, debug, and stacktrace logging
+    #>
+    [CmdletBinding()]
+    param(
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $PlanName,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $ModulePath,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $BoltDir,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $ConfigFile,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [bool]    $Color=$true,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [BoltOutputLevel]  $OutputLevel
+    )
+    begin {
+        $_flags = Convert-BoltParams @PSBoundParameters
+    }
+    process {
+        if ($OutputLevel) {
+            $_result = Invoke-Expression "bolt plan show $PlanName --format=json $_flags"
+        } else {
+            $_result = Invoke-Expression "bolt plan show $PlanName --format=json $_flags" 2>$null
+        }
+        return ($_result | ConvertFrom-Json)
+    }
+}
+
+function Read-BoltTasks {
```

Comment:
Should be singular, not plural

Reply:
Actually, in retrospect this should really be `Get-BoltTask` not read

https://docs.microsoft.com/en-us/powershell/developer/cmdlet/approved-verbs-for-windows-powershell-commands#get-vs-read

Reply:
One object per module on the output pipeline